From 1ece5bfd00eda071ff19d17035d1af23f4d160e9 Mon Sep 17 00:00:00 2001
From: Thomas Huth <thuth@redhat.com>
Date: Mon, 9 Oct 2017 15:50:34 +0200
Subject: [PATCH 31/34] RHEL: Add RHEL7 machine type for qemu on s390x

RH-Author: Thomas Huth <thuth@redhat.com>
Message-id: <1507564234-19627-3-git-send-email-thuth@redhat.com>
Patchwork-id: 77037
O-Subject: [RHEL-7.5 qemu-kvm-ma PATCH 2/2] RHEL: Add RHEL7 machine type for qemu on s390x
Bugzilla: 1473292
RH-Acked-by: Cornelia Huck <cohuck@redhat.com>
RH-Acked-by: Jens Freimann <jfreimann@redhat.com>
RH-Acked-by: Laurent Vivier <lvivier@redhat.com>

Upstream-status: downstream only

Since downstream qemu-kvm is an engineered version (with additional
bug fixes and features on top of the upstream version), we can not
use the upstream machine types here. Introduce RHEL-specific machine
types instead.

Signed-off-by: Thomas Huth <thuth@redhat.com>
Signed-off-by: Miroslav Rezanina <mrezanin@redhat.com>
---
 hw/s390x/s390-virtio-ccw.c | 17 ++++++++++++++++-
 1 file changed, 16 insertions(+), 1 deletion(-)

diff --git a/hw/s390x/s390-virtio-ccw.c b/hw/s390x/s390-virtio-ccw.c
index 1aec58c..4e524b0 100644
--- a/hw/s390x/s390-virtio-ccw.c
+++ b/hw/s390x/s390-virtio-ccw.c
@@ -411,7 +411,7 @@ bool css_migration_enabled(void)
     {                                                                         \
         MachineClass *mc = MACHINE_CLASS(oc);                                 \
         ccw_machine_##suffix##_class_options(mc);                             \
-        mc->desc = "VirtIO-ccw based S390 machine v" verstr;                  \
+        mc->desc = "VirtIO-ccw based S390 machine " verstr;                   \
         if (latest) {                                                         \
             mc->alias = "s390-ccw-virtio";                                    \
             mc->is_default = 1;                                               \
@@ -435,6 +435,8 @@ bool css_migration_enabled(void)
     }                                                                         \
     type_init(ccw_machine_register_##suffix)
 
+#if 0 /* Disabled for Red Hat Enterprise Linux */
+
 #define CCW_COMPAT_2_9 \
         HW_COMPAT_2_9 \
         {\
@@ -605,6 +607,19 @@ static void ccw_machine_2_4_class_options(MachineClass *mc)
 }
 DEFINE_CCW_MACHINE(2_4, "2.4", false);
 
+#else
+
+static void ccw_machine_rhel750_instance_options(MachineState *machine)
+{
+}
+
+static void ccw_machine_rhel750_class_options(MachineClass *mc)
+{
+}
+DEFINE_CCW_MACHINE(rhel750, "rhel7.5.0", true);
+
+#endif
+
 static void ccw_machine_register_types(void)
 {
     type_register_static(&ccw_machine_info);
-- 
1.8.3.1

